src/main/java/com/aws/greengrass/status/FleetStatusService.java

```diff
@@ -138,10 +138,16 @@ public FleetStatusService(Topics topics, MqttClient mqttClient, DeploymentStatus
                               Kernel kernel, DeviceConfiguration deviceConfiguration,
                               PlatformResolver platformResolver, int periodicUpdateIntervalSec) {
         super(topics);
-
         this.mqttClient = mqttClient;
         this.deploymentStatusKeeper = deploymentStatusKeeper;
         this.kernel = kernel;
+        this.deviceConfiguration = deviceConfiguration;
+
+        if (!deviceConfiguration.isDeviceConfiguredToTalkToCloud()) {
+            logger.atWarn().log("Device not configured to talk to AWS Iot cloud. FSS is offline");
+            return;
```

Comment:
Same here as with the MQTT client. Don't partially configure the class, construct the whole thing, but modify its behavior after construction.